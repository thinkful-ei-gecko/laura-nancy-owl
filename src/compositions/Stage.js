import React from 'react';
import './Stage.css';

function Stage(props){

    const onStage = props.participants.filter(participant => (participant.inSession && participant.onStage)).map(
        participant => (
            <div key={participant.id}>
                {participant.name}
                <img src={participant.avatar} alt="{participant.name}"/>
            </div>
        ))
    return (
        <section className='stage'>
            {onStage}
        </section>
    )
}

export default Stage;