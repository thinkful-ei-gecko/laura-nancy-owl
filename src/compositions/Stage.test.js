import React from 'react';
import ReactDOM from 'react-dom';
import Stage from './Stage';
import participants from './ParticipantsList';
import renderer from 'react-test-renderer';

describe('<Stage />', () => {
    it('renders without crashing', () => {
        // Create a DOM element to render the component into
        const div = document.createElement('div');

        // Render the component
        // If something is wrong it will fail here
        ReactDOM.render(<Stage participants = {participants}/>, div);

        // Clean up
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders this UI as expected', () => {
        // Render the component, as JSON
        const tree = renderer.create(<Stage participants = {participants}/>).toJSON();
        // Check whether it matches the previous snapshot
        // Stored in __snapshots__/App.test.js.snap
        expect(tree).toMatchSnapshot(); 
    });
});