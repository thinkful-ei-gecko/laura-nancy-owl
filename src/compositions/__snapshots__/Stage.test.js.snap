// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stage /> renders this UI as expected 1`] = `
<section
  className="stage"
>
  <div>
    Koren Templeton
    <img
      alt="{participant.name}"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div>
    Ashla Attwool
    <img
      alt="{participant.name}"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
  </div>
</section>
`;
